### PR TITLE
Cross checkbox implementation

### PR DESCRIPTION
**Pull request by M4dhav**
Implemented feature as required in #101

### CODE REVIEW

File: lib/presentation/atoms/strikethrough_tile_atom.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import 'package:animated_line_through/animated_line_through.dart';
+import 'package:flutter/material.dart';
+import 'package:infinite_horizons/presentation/atoms/atoms.dart';
+
+class StrikethroughTileAtom extends StatelessWidget {
```
Comment (guyluz11) [inline lib/presentation/atoms/strikethrough_tile_atom.dart:5]: This should be variant of ListTileAtom

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 2]
```diff
@@ -46,8 +46,9 @@ class _TipsOrganismState extends State<TipsOrganism> {
     });
   }
 
-  void onCheckBox(int id, bool value) =>
-      StudyTypeAbstract.instance!.setTipValue(id, value);
+  void onCheckBox(int id, bool value) {
+    StudyTypeAbstract.instance!.setTipValue(id, value);
+  }
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:51]: No reason for this change

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: lib/presentation/atoms/list_tile_atom.dart [hunk 3]
```diff
@@ -21,19 +24,46 @@ class ListTileAtom extends StatelessWidget {
...
+              : TextAtom(subtitle!, translate: translateSubtitle),
+          leading: leading,
+          trailing: trailing,
+          onTap: onTap,
+        );
+      case ListTileVariant.strikethrough:
+        return ListTile(
+          enabled: enable,
+          contentPadding: EdgeInsets.zero,
+          title: AnimatedLineThrough(
```
Comment (guyluz11) [inline lib/presentation/atoms/list_tile_atom.dart:49]: Please move AnimatedLineThrough to a separate atom, we will implement the sound effect in the future inside of the atoms

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: lib/presentation/atoms/list_tile_atom.dart [hunk 4]
```diff
@@ -21,19 +24,46 @@ class ListTileAtom extends StatelessWidget {
...
-      subtitle: subtitle == null
-          ? null
-          : TextAtom(subtitle!, translate: translateSubtitle),
-      leading: leading,
-      trailing: trailing,
-      onTap: onTap,
-    );
+    switch (variant) {
+      case ListTileVariant.regular:
+        return ListTile(
```
Comment (guyluz11) [inline lib/presentation/atoms/list_tile_atom.dart:34]: There is no reason to duplicate the code for each case.
Make var of the subtitle and inset it in the end

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: lib/presentation/atoms/list_tile_atom.dart (hunk 4 above)
Comment (M4dhav) [reply lib/presentation/atoms/list_tile_atom.dart:34]: Please elaborate, I am unable to understand what you mean

Comment (guyluz11) [reply lib/presentation/atoms/list_tile_atom.dart:34]:
```

  @override
  Widget build(BuildContext context) {

  Widget title ;
    switch (variant) {
      case ListTileVariant.regular:
        title = TextAtom(title, translate: translateTitle)
      case ListTileVariant.strikethrough:
        title = AnimatedLineThrough(
          duration: const Duration(milliseconds: 300),
          strokeWidth: 2,
          isCrossed: isCrossed,
          child: TextAtom(title, translate: translateTitle),
        );
    }

      return ListTile(
          enabled: enable,
          contentPadding: EdgeInsets.zero,
          title: title,
          subtitle: subtitle == null
              ? null
              : TextAtom(subtitle!, translate: translateSubtitle),
          leading: leading,
          trailing: trailing,
          onTap: onTap,
        );
  }
```

File: lib/presentation/atoms/list_tile_atom.dart [hunk 5]
```diff
@@ -21,19 +24,46 @@ class ListTileAtom extends StatelessWidget {
...
+          leading: leading,
+          trailing: trailing,
+          onTap: onTap,
+        );
+    }
   }
 }
+
+enum ListTileVariant {
+  regular,
```
Comment (guyluz11) [inline lib/presentation/atoms/list_tile_atom.dart:67]: This name does not contain any meaning of what it is.
Change to text

File: lib/presentation/atoms/list_tile_atom.dart [hunk 6]
```diff
@@ -21,19 +24,46 @@ class ListTileAtom extends StatelessWidget {
...
+              : TextAtom(subtitle!, translate: translateSubtitle),
+          leading: leading,
+          trailing: trailing,
+          onTap: onTap,
+        );
+    }
   }
 }
+
+enum ListTileVariant {
```
Comment (guyluz11) [inline lib/presentation/atoms/list_tile_atom.dart:66]: Maybe ListTileSubtitleVariant is more appropriate since this is the only change

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by guyluz11** [comment]
Hey @M4dhav, I see that this pr hasn't changed from last week.

I would love to add it to the next version, this is a p1 issue.
Can you estimate when you will return to it?.

**Comment by M4dhav** [comment]
Hey, I have been busy for the past week, I will try to get it done in 2/3 days

**Comment by guyluz11** [comment]
> Hey, I have been busy for the past week, I will try to get it done in 2/3 days

Thanks for the update

**Pull request closed without merging**